src/components/Modals/TaskModal.js

```diff
@@ -0,0 +1,66 @@
+import Modal from "react-bootstrap/Modal";
+import Button from "react-bootstrap/Button";
+import { useState } from "react";
+import taskStore from "../../stores/taskStore";
+import { observer } from "mobx-react";
+import "../../App.css";
+
+const TaskModal = () => {
+  const [show, setShow] = useState(false);
+  const toggleShow = () => setShow(!show);
+
+  const [task, setTask] = useState({
+    name: "",
+    description: "",
+    status: false,
+    priority: "low",
+  });
+
+  const handleChange = (e) => {
+    setTask({ ...task, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    taskStore.addTask(task);
+  };
+
+  return (
+    <div className="Modal">
+      <Button variant="primary" onClick={toggleShow}>
```

Comment:
Remove ``onclick={toggleShow}`` and call ``toggleShow()`` in handle submit. This button will be a part of the form.